### PR TITLE
[2131] Pen test environment

### PR DESCRIPTION
**Pull request by saliceti**
### Context
Preparation for pen test

### Changes proposed in this pull request
Terraform and app config for pen test environment

### CODE REVIEW

**Comment by saliceti** [comment]
[Set up Register pen test environment](https://trello.com/c/mVGTmwXl/2131-set-up-register-pen-test-environment)

File: terraform/workspace-variables/pen.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export TF_VAR_key_vault_name=s121t01-shared-kv-01
+export TF_VAR_key_vault_app_secret_name=REGISTER-APP-SECRETS-STAGING
```
Comment (vigneshmsft) [inline terraform/workspace-variables/pen.sh:2]: should we create a copy of the app secrets for pen test environment?

File: terraform/workspace-variables/pen.tfvars [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# PaaS
+paas_app_environment       = "pen"
+env_config                 = "pen"
+paas_space_name            = "bat-staging"
+paas_postgres_service_plan = "tiny-unencrypted-11"
```
Comment (vigneshmsft) [inline terraform/workspace-variables/pen.tfvars:5]: restore from snapshot option won't work for tiny plan..
snapshots are only available for small plans and above.

File: terraform/variables.tf [hunk 3]
```diff
@@ -45,6 +45,7 @@ variable key_vault_infra_secret_name {}
 
 variable azure_credentials { default = null }
 
+variable "restore_db_from_instance" { default = null }
```
Comment (vigneshmsft) [inline terraform/variables.tf:48]: both DBs will have to be in the same plan for this work. and restore from snapshot option only works for small plans and above.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -41,12 +45,12 @@ jobs:
       - id:   select-environments
         if:  github.event.inputs.pr == ''
         uses: DFE-Digital/bat-infrastructure/actions/prepare-environment-matrix@main
```
Comment (vigneshmsft) [inline .github/workflows/deploy.yml:47]: need to add `pen` environment in the above [action](https://github.com/DFE-Digital/bat-infrastructure/blob/main/actions/prepare-environment-matrix/action.yml#L7)

**Review by misaka — APPROVED** [review]
👍

File: terraform/workspace-variables/pen.sh (hunk 1 above)
Comment (saliceti) [reply terraform/workspace-variables/pen.sh:2]: As discussed with @misaka , we can use the same secrets as staging

File: terraform/workspace-variables/pen.tfvars (hunk 2 above)
Comment (saliceti) [reply terraform/workspace-variables/pen.tfvars:5]: Thanks, as discussed with @misaka , we will build an empty database instead

**Review by vigneshmsft — APPROVED** [review]

**Pull request merged**